tests/scalapack/scalapack_10_a.cc

```diff
@@ -0,0 +1,114 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#include "../lapack/create_matrix.h"
+#include "../tests.h"
+
+// test saving and loading of distributed ScaLAPACKMatrices
```

Comment:
Could you briefly elaborate what this tests, namely that we had a bug with serial save and parallel load

Reply:
Due to a bug removed by #7044, a matrix that was saved to disk in parallel could not be read correctly in serial and vice versa. The bug was not detected in the ScaLAPACK tests as the save/load functions are just used for matrices with identical process grids and ScaLAPACK blocking factors. The newly added tests save matrices in parallel/serial and afterwards load them in serial/parallel to check the content.

The reason to check for the proper behaviour is that users might create matrices in serial using their private code and then read the data into a ScaLAPACKMatrix for further computations.

Reply:
I think that @davydden wished you to elaborate in the description of the test itself ;-)

Reply:
Ok, no problem!

Reply:
But I don't think we have to elaborate on removed bugs in the description of the test.

Reply:
> I think that @davydden wished you to elaborate in the description of the test itself ;-)

yes, because in 5 years nobody will remember what this tests actually is for and what was the bug it refers to.

Reply:
That's right. We often do that in the form "There was a bug in deal.II whereby we ... when we should have ...  This test verifies that this now actually works." It provides a rationale not only for what the test does, but also why it's there to begin with.